pkg/ovs/ovn-nb-acl.go

```diff
@@ -498,42 +627,51 @@ func (c *ovnClient) CreateBareAcl(parentName, direction, priority, match, action
 // DeleteAcls delete several acl once,
 // delete to-lport and from-lport direction acl when direction is empty, otherwise one-way
 // parentType is 'ls' or 'pg'
-func (c *ovnClient) DeleteAcls(parentName, parentType string, direction string) error {
-	externalIDs := map[string]string{aclParentKey: parentName}
+func (c *ovnClient) DeleteAcls(parentName, parentType string, direction string, externalIDs map[string]string) error {
+	ops, err := c.DeleteAclsOps(parentName, parentType, direction, externalIDs)
+	if err != nil {
+		return err
+	}
 
-	/* delete acls from port group or logical switch */
-	acls, err := c.ListAcls(direction, externalIDs)
+	if err = c.Transact("acls-del", ops); err != nil {
+		return fmt.Errorf("del acls from type %s %s: %v", parentType, parentName, err)
+	}
+
+	return nil
+}
+
+func (c *ovnClient) DeleteAcl(parentName, parentType, direction, priority, match string) error {
+	acl, err := c.GetAcl(parentName, direction, priority, match, true)
 	if err != nil {
-		return fmt.Errorf("list type %s %s acls: %v", parentType, parentName, err)
+		return err
 	}
 
-	aclUUIDs := make([]string, 0, len(acls))
-	for _, acl := range acls {
-		aclUUIDs = append(aclUUIDs, acl.UUID)
+	if acl == nil {
+		return nil // skip if acl not exist
 	}
 
 	var removeAclOp []ovsdb.Operation
 	if parentType == portGroupKey { // remove acl from port group
-		removeAclOp, err = c.portGroupUpdateAclOp(parentName, aclUUIDs, ovsdb.MutateOperationDelete)
+		removeAclOp, err = c.portGroupUpdateAclOp(parentName, []string{acl.UUID}, ovsdb.MutateOperationDelete)
 		if err != nil {
-			return fmt.Errorf("generate operations for deleting acls from port group %s: %v", parentName, err)
+			return fmt.Errorf("generate operations for deleting acl from port group %s: %v", parentName, err)
 		}
 	} else { // remove acl from logical switch
-		removeAclOp, err = c.logicalSwitchUpdateAclOp(parentName, aclUUIDs, ovsdb.MutateOperationDelete)
+		removeAclOp, err = c.logicalSwitchUpdateAclOp(parentName, []string{acl.UUID}, ovsdb.MutateOperationDelete)
 		if err != nil {
-			return fmt.Errorf("generate operations for deleting acls from logical switch %s: %v", parentName, err)
+			return fmt.Errorf("generate operations for deleting acl from logical switch %s: %v", parentName, err)
 		}
 	}
 
-	// delete acls
-	delAclsOp, err := c.WhereCache(aclFilter(direction, externalIDs)).Delete()
+	// delete acl
+	delAclOp, err := c.Where(acl).Delete()
 	if err != nil {
-		return fmt.Errorf("generate operation for deleting acls: %v", err)
+		return fmt.Errorf("generate operation for deleting acl: %v", err)
 	}
 
-	ops := make([]ovsdb.Operation, 0, len(removeAclOp)+len(delAclsOp))
+	ops := make([]ovsdb.Operation, 0, len(removeAclOp)+len(delAclOp))
 	ops = append(ops, removeAclOp...)
-	ops = append(ops, delAclsOp...)
+	ops = append(ops, delAclOp...)
```

Comment:
The `acls` column has a strong reference to the ACL table, so there is no need to delete the ACL.